Src/LiquidProjections/Dispatcher.cs

```diff
@@ -18,27 +20,37 @@ public Dispatcher(IEventStore eventStore)
             this.eventStore = eventStore;
         }
 
-        public void Subscribe(long? checkpoint, Func<IReadOnlyList<Transaction>, Task> handler)
+        public IDisposable Subscribe(long? checkpoint, Func<IReadOnlyList<Transaction>, Task> handler)
         {
             if (handler == null)
             {
                 throw new ArgumentNullException(nameof(handler));
             }
 
-            // TODO: intercept and log errors
+            var subscriptionMonitor = new object();
+            IDisposable subscription = null;
 
-            eventStore.Subscribe(checkpoint, async transactions =>
+            lock (subscriptionMonitor)
             {
-                try
-                {
-                    await handler(transactions);
-                }
-                catch (Exception e)
+                subscription = eventStore.Subscribe(checkpoint, async transactions =>
                 {
-                    Console.WriteLine(e);
-                    throw;
-                }
-            });
+                    try
+                    {
+                        await handler(transactions);
+                    }
+                    catch (Exception exception)
+                    {
+                        ProjectionException.LogException(exception, LogLevel.Fatal);

```

Comment:
I don't think the exception itself should log directly. That's mixing responsibilities